scrapy/core/downloader/handlers/negotiation.py

```diff
@@ -145,8 +125,6 @@ def get_download_handler(self, protocol):
         elif protocol == 'h2':
```

Comment:
Maybe turn this into an `else` and indicate in a comment that the value is expected to be `h2` here? Same in `add_connection`.